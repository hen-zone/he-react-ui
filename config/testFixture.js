import createTestContext from 'react-cosmos-test/enzyme';

export default function testFixture(fixture, variant) {
  const { mount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component;
  mount();
  test(`<${displayName ||
    name} /> rendered correctly with ${variant} fixture`, () =>
    expect(getWrapper()).toMatchSnapshot());
}
