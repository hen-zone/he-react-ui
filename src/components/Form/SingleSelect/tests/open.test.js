import testFixture from '../../../../../config/testFixture';
import fixture from '../fixtures/open.fixture.js';

testFixture(fixture, 'label');
