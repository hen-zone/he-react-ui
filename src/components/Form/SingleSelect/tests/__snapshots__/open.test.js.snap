// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleSelect /> rendered correctly with label fixture 1`] = `
<SingleSelect
  disabled={false}
  error=""
  id="demo"
  inline={false}
  name="demo"
  onBeforeOpen={[Function]}
  onChange={[Function]}
  onClose={[Function]}
  options={
    Array [
      Object {
        "label": "Option 1",
        "value": "1",
      },
      Object {
        "label": "Option 2",
        "value": "2",
      },
      Object {
        "label": "Option 3",
        "value": "3",
      },
    ]
  }
  value={null}
>
  <div
    className="outer"
  >
    <div
      className="relative"
    >
      <div
        className="container expanded"
      >
        <Select
          arrowRenderer={[Function]}
          autosize={true}
          backspaceRemoves={true}
          backspaceToRemoveMessage="Press backspace to remove {label}"
          className="select expanded"
          clearAllText="Clear all"
          clearRenderer={[Function]}
          clearValueText="Clear value"
          clearable={true}
          closeOnSelect={true}
          deleteRemoves={true}
          delimiter=","
          disabled={false}
          escapeClearsValue={true}
          filterOptions={[Function]}
          id="demo"
          ignoreAccents={true}
          ignoreCase={true}
          inputProps={Object {}}
          isLoading={false}
          joinValues={true}
          labelKey="label"
          matchPos="any"
          matchProp="any"
          menuBuffer={0}
          menuRenderer={[Function]}
          multi={false}
          name="demo"
          noResultsText="No results found"
          onBlurResetsInput={true}
          onChange={[Function]}
          onClose={[Function]}
          onCloseResetsInput={true}
          onOpen={[Function]}
          onSelectResetsInput={true}
          openOnClick={true}
          optionComponent={[Function]}
          options={
            Array [
              Object {
                "label": "Option 1",
                "value": "1",
              },
              Object {
                "label": "Option 2",
                "value": "2",
              },
              Object {
                "label": "Option 3",
                "value": "3",
              },
            ]
          }
          pageSize={5}
          placeholder="Select..."
          removeSelected={true}
          required={false}
          rtl={false}
          scrollMenuIntoView={true}
          searchable={true}
          simpleValue={false}
          tabSelectsValue={true}
          trimFilter={true}
          value={null}
          valueComponent={[Function]}
          valueKey="value"
          valueRenderer={[Function]}
        >
          <div
            className="Select select expanded is-clearable is-searchable Select--single"
          >
            <input
              disabled={false}
              name="demo"
              type="hidden"
              value=""
            />
            <div
              className="Select-control"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
            >
              <span
                className="Select-multi-value-wrapper"
                id="react-select-2--value"
              >
                <div
                  className="Select-placeholder"
                >
                  Select...
                </div>
                <AutosizeInput
                  aria-activedescendant="react-select-2--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  className="Select-input"
                  id="demo"
                  injectStyles={true}
                  minWidth="5"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  value=""
                >
                  <div
                    className="Select-input"
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-activedescendant="react-select-2--value"
                      aria-expanded="false"
                      aria-haspopup="false"
                      aria-owns=""
                      id="demo"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      role="combobox"
                      style={
                        Object {
                          "boxSizing": "content-box",
                          "width": "5px",
                        }
                      }
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    />
                  </div>
                </AutosizeInput>
              </span>
              <span
                className="Select-arrow-zone"
                onMouseDown={[Function]}
              >
                <span
                  className="Select-arrow"
                  onMouseDown={[Function]}
                />
              </span>
            </div>
          </div>
        </Select>
        <DropDown
          className="caret"
          fill="currentColor"
          height={32}
          width={32}
        >
          <svg
            className="caret"
            fill="currentColor"
            height={32}
            viewBox="0 0 24 24"
            width={32}
          >
            <path
              d="M8.068 9h7.864a.5.5 0 0 1 .385.82l-3.933 4.72a.5.5 0 0 1-.768 0L7.683 9.82A.5.5 0 0 1 8.068 9z"
            />
          </svg>
        </DropDown>
      </div>
    </div>
  </div>
</SingleSelect>
`;
